### DISCUSSION TITLE
Embedded images in issues, pull requests and comments won't open full-size (404 response)

### DISCUSSION THREAD

**Discussion by zbjornson** [community/community#59497] [Projects and Issues]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

Lately, images in issues, comments and PRs sometimes fail to load when you click on them to open them full-size. The URL looks like this:

`https://private-user-images.githubusercontent.com/12348/123456-9e9ae87f-006e-4394-91d7-94c134e0f9cb.png?jwt=[something]`

and the request gets a 404 response.

It's only happening intermittently, but when it happens, it's happening for my coworkers in addition to myself.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for reporting this @zbjornson - do you and/or your coworkers mind sharing some additional details that may help the team as they investigate like browser type(s), image type(s), and the size of the images? Appreciate it.

**Reply by zbjornson** [reply]
Thanks for the follow-up. I was in Firefox 114, my coworker Safari 16.4.1. Images are PNGs, and ranged from 100x500 to 800x800 px. I'm not sure if it was coincidence or not, but the issue went away when I edited the comments with the images, even though I didn't change the images.

**Comment by OutOfSite-Support** [comment]
Seems to be happening for me as well. Any indication on why this is happening or solutions found already?

**Reply by queenofcorgis** [reply]
@OutOfSite-Support could you share your browser type(s), image type(s), and the size of the images?

**Reply by OutOfSite-Support** [reply]
Here is some details.
BTW. It seems the "jwt" parameter in the url does change ever so slightly with just a handful of characters, when you are on a page for a long time (i can not tell you how long, most likely 10+ minutes).
If you do not refresh the page before replying to someone, it will still use the old "jwt" token it seems to have saved.
When you refresh the page before replying, it will generate a new "jwt" token (very slight differences) and this one does seem to work properly.
It also seems to only happen with private images where the "jwt" parameter seems to be needed, that's why I am talking about the parameter here.

important to know; I am talking here about a token, but I have no clue what the value of the "jwt" parameter actually is or does within github, to me it seems like some sort of token.

t.b.h. I don't think it has anything to do with the image or browser, seems to me the way that github saves and retrieves an image is the problem here based on my observations, but I could be completely wrong ofcourse.

Here is some details about my browser and the image;

Google Chrome
Versie 115.0.5790.171 (Officiële build) (64-bits)

Editie	Windows 11 Pro
Versie	21H2
Installatiedatum	‎14-‎11-‎2022
Build van besturingssysteem	22000.2176
Ervaring	Windows Feature Experience Pack 1000.22001.1000.0

![image](https://github.com/community/community/assets/121167153/6ff652c8-05d3-4dee-b5de-261298ff6205)

**Comment by DanielGibson** [comment]
This behavior still persists.
If an issue has been open for too long in a browsertab (and "too long" isn't very long, probably less than an hour), clicking an image that was attached to a post and integrated into the posts markdown opens a new tab with an empty page (at least in Firefox?). 
You get a 404 there, but as it's for an image file and not an html file, you don't get a website that actually says 404 but just an empty page.

I don't know what a timeout for images embedded into posts is supposed to achieve - *one would think that them having a stable URL makes it easier to cache them both on your server and in the users browsercache so they don't have to be downloaded again and again* - but this is very annoying.

**Reply by DanielGibson** [reply]
Actually, it's only **five minutes**, I just tried it.

@queenofcorgis to reproduce:
1. Open an issue or PR or discussion thread that contains images (that are attached to the post and integrated in markdown), like this one, in a desktop browser. I tested this with latest Firefox and Chromium on XUbuntu Linux 24.04
2. Click one of the images and notice how it opens in a new browser tab.
3. Wait for at least 5 minutes without closing the browser tab with the issue/PR/discussion thread
4. Click an image again. A new tab will open, but this time it will either be blank (Firefox) or show an error page (Chromium - "This private-user-images.githubusercontent.com page can’t be found")
   - If the new tab *does* show the image, try refreshing it - it might have shown it from the browser cache, especially if it's the same image you clicked in step 2.

**Reply by DanielGibson** [reply]
Imagine how much *fun* this is if it there are many replies and you had to click "Load more…" several times to see them all and now you gotta reload because five minutes are over and you can't view images at their full size anymore - and afterwards click "Load more…" several times again to see all posts.
(That hidden items "feature" [is pretty annoying too](https://github.com/orgs/community/discussions/130618), by the way.)

**Reply by zimmerman-pliant** [reply]
Yes, I came here because I was having this problem, and I can confirm that I can reproduce this.

**Reply by zimmerman-pliant** [reply]
Also can confirm that five minutes is enough

**Reply by bonaparte89** [reply]
Many people in our company have the same issue. You always have to refresh the page and then scroll back to open it..
Also videos dont play in brave. I always have to use edge... 
I dont get why the JWT isnt silently refreshed in the background?
